rclpy/guard_conditions/examples_rclpy_guard_conditions/trigger_guard_condition.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import time
+
+import rclpy
+from rclpy.executors import GuardCondition
+
+
+def main(args=None):
+
+    rclpy.init(args=args)
+    node = rclpy.create_node('DemoGuardCondition')
```

Comment:
nit: for consistency with other demo nodes, I'd use *snake_case* for the node name.